pkg/scheduler/scheduler.go

```diff
@@ -505,19 +504,11 @@ func (s *Scheduler) patchScheduleResultForResourceBinding(oldBinding *workv1alph
 	newBinding.Annotations[util.PolicyPlacementAnnotation] = placement
 	newBinding.Spec.Clusters = scheduleResult
 
-	oldData, err := json.Marshal(oldBinding)
+	patchBytes, err := helper.GenMergePatch(oldBinding, newBinding)
```

Comment:
Why did we use `application/merge-patch+json` to patch the resource instead of using `application/json-patch+json`? 

`application/json-patch+json` has a smaller data exchange.

Reply:
It's not absolute. For example:
data: `{"name":"foo", "class":"A", "count":3}`
We want to update `name` and `class` fields.
* Use _JSON Patch_: 
```
[
	{
		"op" : "replace" ,
		"path" : "/name" ,
		"value" : "bar"
	},
	{
		"op" : "replace" ,
		"path" : "/class" ,
		"value": "B"
        }
]
```

* Use _JSON Merge Patch_: 
```
{"name":"foo", "class":"A"}
```
Both have advantages and disadvantages, and none of them will fit everybody’s usecases, but I think _JSON Merge 
 Patch_ is enough here and more simple.